#/usr/bin/env python3
PACKAGE = "usb_cam"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("brightness", int_t, 0, "brightness", 0, -64, 64)
gen.add("contrast", int_t, 0, "contrast", 32, 0, 64)
gen.add("saturation", int_t, 0, "saturation", 64, 0, 128)
gen.add("gain", int_t, 0, "gain", 0,  0, 100)
gen.add("sharpness", int_t, 0, "sharpness", 2, 0, 6)
gen.add("backlight_compensation", int_t, 0, "backlight_compensation", 1, 0, 2)

gen.add("auto_white_balance", bool_t, 0, "auto_white_balance", True)
gen.add("white_balance", int_t, 0, "white_balance", 4600, 2800, 6500)

gen.add("autoexposure", int_t, 0, "autoexposure",  3, 0, 3)
gen.add("exposure", int_t, 0, "exposure", 157, 1, 5000)

exit(gen.generate(PACKAGE, "usb_cam", "usb_cam_param"))
